modules/ROOT/pages/sources-reference.adoc

```diff
@@ -0,0 +1,78 @@
+include::_attributes.adoc[]
+endif::[]
+:keywords: anypoint studio, flow, source, sources, redelivery, policy
+
+Sources are a very important part of any Mule Application. When added to a flow, they serve as the entry point for information,
+linking any external communication protocol with internal Mule processing.
+
+Examples of sources are:
+
+* xref:docs-connectors::http-connector.adoc[HTTP:Listener]
+* xref:docs-connectors::file-connector.adoc[FILE:Directory Listener]
+* xref:docs-connectors::amqp-connector.adoc[AMQP:Consume]
+
+== Configuring Sources
+
+Even though each source implementation will have different configuration parameters, there are some common ones worth mentioning.
+
+=== Redelivery Policy
+
+Each source can be configured with a redelivery policy. That way, an user can add some logic on how to handle messages that arrive at the source multiple times and cause errors.
+
+[IMPORTANT]
+Some sources implemented with the xref:docs-mule-sdk::index.adoc[Mule SDK] have their own logic for a redelivery policy. In those cases, check the documentation for the connector which that source belongs to for more information.
+
+For example:
+[source,xml]
+----
+<http:listener path="exhaustRedelivery" config-ref="listenerConfig">
+    <redelivery-policy maxRedeliveryCount="10" idExpression="#[payload.transaction.id]"/>
+</http:listener>
+----
+
+==== Redelivery Policy Configuration Parameters
+
+[%header,cols="20a,20a,20a,60a"]
+|===
+|Field |Type |Default Value |Description
+|*Max Redelivery Count* |Number |5 |The maximum number of times a message can be redelivered and processed unsuccessfully before raising a `MULE:REDELIVERY_EXHAUSTED` error.
+|*Use Secure Hash* |Boolean |True |Whether to use a secure hash algorithm to identify a redelivered message.
+|*Message Digest Algorithm* |String |SHA-256 |The secure hashing algorithm to use.
+|*ID Expression* |Expression|- | Defines one or more expressions to use to determine when a message has been redelivered. This property may only be set if useSecureHash is false.
+|*Object Store* |String |An Object Store created by the ObjectStoreManager. Non-persistent, with an entryTtl of 300 SECONDS and an expiration interval of 6 SECONDS |The object store where the redelivery counter for each message is going to be stored. It can be configured as a reference or as an inner element. See xref:connectors::object-store/object-store-connector.adoc[Object Store connector]
+|===
+
+==== Redelivery policy functionality
+
+The redelivery policy configured will work as follows:
+
+Each time a new message is received by the source, it will be digested (A key will be generated from the message), either by using the configured Message Digest Algorithm or by evaluating the ID Expression. +
+If the message causes an exception, a counter associated with it will be incremented. Once the counter reaches a value bigger than the configured maxRedeliveryCount, an `MULE:REDELIVERY_EXHAUSTED` error will be thrown. +
```

Comment:
```suggestion
If the message causes an exception, a counter associated with it will be incremented. Once the counter reaches a value bigger than the configured maxRedeliveryCount, a `MULE:REDELIVERY_EXHAUSTED` error will be thrown. +
```